src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_IAwaitExpression.vb

```diff
@@ -211,5 +211,78 @@
 
             VerifyOperationTreeAndDiagnosticsForTest(Of ExpressionStatementSyntax)(source, expectedOperationTree, expectedDiagnostics, useLatestFramework:=True)
         End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67616")>
+        Public Sub TestAwaitExpression_InStatement()
+            Dim source = <![CDATA[
+Imports System.Threading.Tasks
+
+Public Module Program
+    Public Async Function M() As Task(Of Integer)
+        Await M2()'BIND:"Await M2()"
+        Return 0
+    End Function
+
+    Public Function M2() As Task(Of String)
+        Throw New System.Exception()
+    End Function
+End Module
+]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IAwaitOperation (OperationKind.Await, Type: System.String) (Syntax: 'Await M2()')
+  Expression:
+    IInvocationOperation (Function Program.M2() As System.Threading.Tasks.Task(Of System.String)) (OperationKind.Invocation, Type: System.Threading.Tasks.Task(Of System.String)) (Syntax: 'M2()')
+      Instance Receiver:
+        null
+      Arguments(0)
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of AwaitExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics, useLatestFramework:=True)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67616")>
+        Public Sub TestAwaitExpression_InStatement_InSubLambda()
+            Dim source = <![CDATA[
+Imports System
+Imports System.Threading.Tasks
+
+Public Module Program
+    Public Async Function M() as Task
+        Dim lambda As Action = Async Sub()
+                                 Await M2()'BIND:"Await M2()"
+                               End Sub
+
+        Await lambda()
+    End Function
+
+    Public Function M2() As Task(Of String)
+        Throw New System.Exception()
+    End Function
+End Module
+]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IAwaitOperation (OperationKind.Await, Type: System.String) (Syntax: 'Await M2()')
+  Expression:
+    IInvocationOperation (Function Program.M2() As System.Threading.Tasks.Task(Of System.String)) (OperationKind.Invocation, Type: System.Threading.Tasks.Task(Of System.String)) (Syntax: 'M2()')
+      Instance Receiver:
+        null
+      Arguments(0)
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC30491: Expression does not produce a value.
```

Comment:
Is there a benefit in having this error?

Reply:
I think we actually want to verify that we can emit and run the code.

Reply:
This scenario hasn't changed. We already produce an error today, so I'm not sure what you're asking.

Reply:
> This scenario hasn't changed. We already produce an error today, so I'm not sure what you're asking.

I am asking why you are testing this error scenario. An action cannot be awaited, and it really doesn't matter what is in the lambda that was used to create the Action. The error doesn't test anything related to the change you are making.

Reply:
So, unless you have a specific reason to test this error scenario in connection to the change that you are making, consider getting rid of the error and converting the test into a success form (build/execute) similar to the test below.